ServerCore/Pages/Puzzles/Checklist.cshtml.cs

```diff
@@ -91,22 +91,18 @@ public async Task<IActionResult> OnGetAsync()
                     int totalDiscount = 0;
 
                     hintsCountThisPuzzle = hints.Count();
+
+                    // positive hints cost what they say they cost.
+                    // negative hints apply discounts to each other, find the max negative hint.
                     foreach (int cost in hints)
                     {
                         totalDiscount = Math.Min(totalDiscount, cost);
+                        totalHintCostThisPuzzle += Math.Max(0, cost);
                     }
 
-                    // totalDiscount is 0 or negative. Start with that cost (flipped to positive
+                    // totalDiscount is 0 or negative. Add that cost (flipped to positive)
                     // as it must be paid in order to reduce the cost of the others.
-                    totalHintCostThisPuzzle = -totalDiscount;
-
-                    foreach (int cost in hints)
-                    {
-                        // Negative cost hints will be ignored because Max(0,negative) is 0.
-                        // Positive cost hints will only be counted for the cost above the discount.
-                        // (The discount is counted against each other hint.)
-                        totalHintCostThisPuzzle += Math.Max(0, cost + totalDiscount);
-                    }
+                    totalHintCostThisPuzzle += -totalDiscount;
```

Comment:
idk if it's more readable or also syntactically safer in case we screw up elsewhere in the future since we don't have tests to use `Math.Abs`